selfservice/hook/web_hook.go

```diff
@@ -332,40 +343,30 @@ func (e *WebHook) execute(ctx context.Context, data *templateContext) error {
 			if canInterrupt || parseResponse {
 				if err := parseWebhookResponse(resp, data.Identity); err != nil {
 					span.SetStatus(codes.Error, err.Error())
-					errChan <- err
+					return err
 				}
 			}
-			errChan <- fmt.Errorf("webhook failed with status code %v", resp.StatusCode)
-			return
+			return fmt.Errorf("webhook failed with status code %v", resp.StatusCode)
 		}
 
 		if parseResponse {
 			if err := parseWebhookResponse(resp, data.Identity); err != nil {
 				span.SetStatus(codes.Error, err.Error())
-				errChan <- err
+				return err
 			}
 		}
 
-		errChan <- nil
-	}()
-
-	if ignoreResponse {
-		traceID, spanID := span.SpanContext().TraceID(), span.SpanContext().SpanID()
-		logger := e.deps.Logger().WithField("otel", map[string]string{
-			"trace_id": traceID.String(),
-			"span_id":  spanID.String(),
-		})
-		go func() {
-			if err := <-errChan; err != nil {
-				logger.WithField("duration", time.Since(startTime)).WithError(err).Warning("Webhook request failed but the error was ignored because the configuration indicated that the upstream response should be ignored.")
-			} else {
-				logger.WithField("duration", time.Since(startTime)).Info("Webhook request succeeded")
-			}
-		}()
 		return nil
 	}
 
-	return <-errChan
```

Comment:
This always waits for one message on the channel, I don't understand how it could even run async as this is blocking AND racy with the go routine also reading from that channel :sweat_smile:

Reply:
Ahh it would return in the `if` above...